AutoRest/Generators/Ruby/Ruby.Tests/RspecTests/parameter_grouping_spec.rb

```diff
@@ -0,0 +1,109 @@
+    required_parameters.body = @body
+    required_parameters.custom_header = @header
+    required_parameters.query = @query
+    required_parameters.path = @path
+
+    result = @client.parameter_grouping.post_required_async(required_parameters).value!
+
+    expect(result.response.status).to eq(200)
+  end
+
+  it 'should accept required parameters but null optional parameters' do
+    required_parameters = ParameterGroupingPostRequiredParameters.new
+    required_parameters.body = @body
+    required_parameters.path = @path
+
+    result = @client.parameter_grouping.post_required_async(required_parameters).value!
+
+    expect(result.response.status).to eq(200)
+  end
+
+  it 'should reject required parameters with missing required property' do
+    required_parameters = ParameterGroupingPostRequiredParameters.new
+    required_parameters.path = @path
+
+    expect { @client.parameter_grouping.post_required_async(required_parameters).value! }.to raise_error(MsRest::ValidationError)
+  end
+
+  it 'should reject null required parameters' do
+    required_parameters = ParameterGroupingPostRequiredParameters.new
+    required_parameters.path = nil
+
+    expect { @client.parameter_grouping.post_required_async(required_parameters).value! }.to raise_error(MsRest::ValidationError)
+  end
+
+  it 'should accept valid optional parameters' do
+    optional_parameters = ParameterGroupingPostOptionalParameters.new
+    optional_parameters.custom_header = @header
+    optional_parameters.query = @query
+
+    result = @client.parameter_grouping.post_optional_async(optional_parameters).value!
+    expect(result.response.status).to eq(200)
+  end
+
+  it 'should accept null optional parameters' do
+    result = @client.parameter_grouping.post_optional_async(nil).value!
+    expect(result.response.status).to eq(200)
+  end
+
+  it 'should allow multiple parameter groups' do
+    first_parameter_group = FirstParameterGroup.new
+    second_parameter_group = ParameterGroupingPostMultiParamGroupsSecondParamGroup.new
+
+    first_parameter_group.header_one = @header
+    first_parameter_group.query_one = @query
+
+    second_parameter_group.header_two = 'header2'
+    second_parameter_group.query_two = 42
+
+    result = @client.parameter_grouping.post_multi_param_groups_async(first_parameter_group, second_parameter_group).value!
+    expect(result.response.status).to eq(200)
+  end
+
+  it 'should allow multiple parameter groups with some defaults omitted' do
+    first_parameter_group = FirstParameterGroup.new
+    second_parameter_group = ParameterGroupingPostMultiParamGroupsSecondParamGroup.new
+
+    first_parameter_group.header_one = @header
+
+    second_parameter_group.query_two = 42
+
+    result = @client.parameter_grouping.post_multi_param_groups_async(first_parameter_group, second_parameter_group).value!
+    expect(result.response.status).to eq(200)
+  end
+
+  it 'should allow parameter group objects to be shared between operations' do
```

Comment:
in this case, shouldn't we be creating the FirstParameterGroup grouping once and pass it to 2 different operations, so test the "sharing" of the parameter group? If not, it looks like this test is validating the same as some of the above.